### DISCUSSION TITLE
CouchDB memory usage

### DISCUSSION THREAD

**Discussion by viperML** [vrtmrz/obsidian-livesync#340] [Q&A]
Hello, thanks for this fantastic extension.

Just wanted to ask to the people who self-host couchdb, how much memory are you using?
After setting it and adding some documents, CouchDB is using **1716MiB** of memory!

**Comment by dlphnkck** [comment]
Is it memory usage and not volume usage?
If memory usage is 1716MiB in my environment, it may cause out of memory.

- Number of files: 9000
- File size: 200MiB
- Use "[deploy_couchdb_to_flyio_v2_with_swap.ipynb (github.com)](https://gist.github.com/vrtmrz/37c3efd7842e49947aaaa7f665e5020a)" to deploy CouchDB.
- During some of the last 7 days, "Perform Cleaning" was executed.

![2023-11-29-Fly io-Memory_Usage](https://github.com/vrtmrz/obsidian-livesync/assets/117900337/9a331b07-c957-4941-96ef-de4725427480)

- Memory total: 217.5MiB 
- Average memory used: 95MiB

**Reply by viperML** [reply]
> Is it memory usage and not volume usage?
Yeah, memory:

```
$ docker stats d22a4a82b487 --no-stream
CONTAINER ID   NAME                                       CPU %     MEM USAGE / LIMIT    MEM %     NET I/O           BLOCK I/O         PIDS
d22a4a82b487   run-1a6103c9-2dfc-128d-3b43-69f9f3cdea4c   0.31%     1.592GiB / 2.93GiB   54.35%    2.23MB / 1.47MB   97.8MB / 16.6MB   35
```

When I set the limit to 1G, it just OOM's. Other sources suggested that perhaps Erlang's OTP is not detecting properly the ambient memory limits, and just trying to maximise the hosts limits (23GiB), although I haven't checked if this holds true.

**Comment by viperML** [comment]
I was using hashicorp's nomad + the docker runner. After moving to plain systemd execution, the memory usage is back to a normal `98.1M`. I have no idea what was wrong with my previous setup.

**Reply by avephill** [reply]
I'm having trouble with this. I'm self-hosting couchdb with the default couchdb image, and it seems to use 1.55GB no matter what I do. I looked in the couchdb docs and added `attachment_stream_buffer_size = 512` to the `local.ini` but that didn't seem to do anything. I also adjusted batch sizes on my client devices.
 What do you mean by plain systemd execution? Should run it baremetal ?

**Reply by viperML** [reply]
https://github.com/viperML/neoinfra/blob/a1607f4b6832395126ca9ce4b079b1a877a3b717/modules/obsidian/default.nix#L99-L107

**Reply by viperML** [reply]
Note that I have moved to Notion

**Reply by avephill** [reply]
Great, thanks for the response, I'll try to get it working!
